packages/ui-library/src/components/options-menu/index.tsx

```diff
@@ -39,14 +39,132 @@ type MenuOption = {
   color?: ButtonProps['color'];
 };
 
-type OptionsMenuOption = DividerOption | MenuOption;
+type BaseMenuOption = DividerOption | MenuOptionWithOptions;
+
+type MenuOptionWithOptions = MenuOption & {
+  options?: BaseMenuOption[];
+};
+
+type OptionsMenuOption = DividerOption | MenuOptionWithOptions;
 
 interface OptionsMenuItemsProps {
   options: OptionsMenuOption[];
   handleClose: () => void;
   anchorEl: HTMLElement | null;
 }
 
+const DividerItem = () => <Divider />;
```

Comment:
What diff does this make?

Reply:
None, just thought it became a bit more legible to have a component type for each type of OptionMenuType, wdyt?

Reply:
Oh i get it. Looks good!